aten/src/ATen/core/function_schema.h

```diff
@@ -147,6 +188,49 @@ struct FunctionSchema {
             is_vararg,
             is_varret) {}
 
+  // check whether this schema is backward compatible with old_schema,
+  // which means all the existing call sites of old_schema can execute with
+  // this schema. we intent to be conservative, and if necessary, we may
+  // relax the checks in future. the following conditions are considered as
+  // this schema is backward compatible with old_schema:
+  //   1) two schemas are equal
+  //   2) two schemas have same number of arguments, and this schema's
+  //      arguments are backward compatible with the corresponding ones in
+  //      argument list of old_schema.
+  //   3) this schema has m argument, old_argument has n argument, m > n.
+  //      the first n arguments of this schema are backward compatible with
+  //      the corresponding arguments of old_schema. the remaning arguments
+  //      must be either OptionalType or provide default values.
+  bool isBackwardCompatibleWith(const FunctionSchema& old_schema) const {
+    if (*this == old_schema) {
```

Comment:
What is the purpose of these equality checks? I am wary because there should be no difference if these are deleted, but they _will_ make it harder to test whether the below code is correct.

Reply:
removed